scipy/stats/tests/test_stats.py

```diff
@@ -2098,6 +2098,7 @@ def test_regress_against_R(self):
         assert_allclose(res.stderr, 0.0519051424731)
         assert_allclose(res.intercept_stderr, 8.0490133029927)
 
+    @pytest.mark.filterwarnings('ignore::DeprecationWarning')
```

Comment:
Please either adapt the test to use the non-deprecated way of passing arguments, or explicitly expect the warning with
```
with pytest.deprecated_call(match=msg):
```

Reply:
@h-vetinari did you see the test above that looks for the warning? Why do we need to look for the warning in all tests?

Reply:
Oh, NM I can probably just remove these. I didn't even check whether we get the warning or not; I just applied the decorator instinctively.

Reply:
Removing these causes the test to fail
```
FAILED scipy/stats/tests/test_stats.py::TestRegression::test_regress_simple_onearg_rows - DeprecationWarning: Inference of the two sets of measurements from a single "argument `x` is deprecated will result in an error in "SciPy 1.16.0; the sets must be specified separately as "`x` and `y`.
FAILED scipy/stats/tests/test_stats.py::TestRegression::test_regress_simple_onearg_cols - DeprecationWarning: Inference of the two sets of measurements from a single "argument `x` is deprecated will result in an error in "SciPy 1.16.0; the sets must be specified separately as "`x` and `y`.
FAILED scipy/stats/tests/test_stats.py::TestRegression::test_regress_shape_error - DeprecationWarning: Inference of the two sets of measurements from a single "argument `x` is deprecated will result in an error in "SciPy 1.16.0; the sets must be specified separately as "`x` and `y`.
```
but IMHO these skips are fine as this test will just be blindly deleted as it is the same as https://github.com/scipy/scipy/blob/e239ba895dd74ead2cedc2ff6782b952c148ee05/scipy/stats/tests/test_stats.py#L2050-L2059
but using the deprecated argument

Reply:
Sure thing. That will make it easier for whomever removes the deprecated behavior.

Reply:
Haha sorry for all of my useless replies. @lucascolley, I think you're right - these tests will just be removed, so I don't see why we should change. I don't think it's worth explicitly checking for the deprecation warning in all of these because I've already done so in `test_one_arg_deprecation`, but I suppose I could remove `test_one_arg_deprecation` and check for the warning here if @h-vetinari would prefer.

Reply:
It's a bad habit to use filterwarnings - they hang around and get forgotten when the warning itself isn't there anymore. We just shouldn't use it unless absolutely necessary. There used to be a [test](https://github.com/scipy/scipy/blob/main/scipy/_lib/tests/test_warnings.py) that explicitly enforced that absence, I guess it has stopped working. 

If you do add filterwarnings, at least leave a comment, e.g. "TODO: drop this test completely once single-arg support is dropped; deprecation warning tested elsewhere".

Reply:
Done.